### PR TITLE
ARROW-13163: [C++][Gandiva] Implement REPEAT function on Gandiva

### PR DESCRIPTION
**Pull request by jpedroantunes**
Implement REPEAT function on Gandiva which concatenate a string "n" times.
- REPEAT(str, int)

### CODE REVIEW

**Comment by jpedroantunes** [comment]
@augustoasilva can you review it please?

File: cpp/src/gandiva/precompiled/string_ops_test.cc [hunk 1]
```diff
@@ -194,6 +194,34 @@ TEST(TestStringOps, TestConvertReplaceInvalidUtf8Char) {
...
+  EXPECT_EQ(std::string(out_str, out_len), "");
+  EXPECT_FALSE(ctx.has_error());
+
+  out_str = repeat_utf8_int32(ctx_ptr, "a", 1, -10, &out_len);
+  EXPECT_EQ(std::string(out_str, out_len), "");
+  EXPECT_THAT(ctx.get_error(),
+              ::testing::HasSubstr("Repeat number can't be negative"));
+  ctx.Reset();
+}
+
```
Comment (augustoasilva) [inline cpp/src/gandiva/precompiled/string_ops_test.cc:224]: Wouldn't it be good to also put a case where you have a value passed that generates a nullptr, thus making the error of not being able to allocate memory?

**Review by augustoasilva — CHANGES_REQUESTED** [review]
For me, it is almost good. In my opinion just add that case that I'm mentioned to the string_ops_test.cc

**Comment by projjal** [comment]
lgtm. Can you fix the checkstyle error

**Review by projjal — APPROVED** [review]

**Pull request closed without merging**